ophyd/_pyepics_shim.py

```diff
@@ -129,13 +134,16 @@ def clear_callbacks(self):
 
 
 def release_pvs(*pvs):
-    for pv in pvs:
-        pv.clear_callbacks()
-        # Perform the clear auto monitor in one of our _dispatcher threads:
-        # they are guaranteed to be in the right CA context
-        wrapped = wrap_callback(_dispatcher, 'monitor', pv.clear_auto_monitor)
-        # queue the call in the 'monitor' _dispatcher:
-        wrapped()
+    # Run _release_pvs in the 'monitor' thread, assuring that the CA context is correct
+    def _release_pvs():
+        for pv in pvs:
+            pv.clear_callbacks()
+            pv.clear_auto_monitor()
+        event.set()
+
+    event = threading.Event()
+    _dispatcher.get_thread_context('monitor').run(_release_pvs)
+    event.wait()
```

Comment:
Should this have a default timeout?

Reply:
It could - but I'm not sure it's necessary here. The only possibility for this to timeout would be in the case of a massively backed-up event queue or deadlock.

In the former case, the caller would want to ensure that the PVs were fully released before proceeding. Without the wait, one could imagine a scenario where there is a race to clear the callbacks here or add callbacks in some other function.